### PR TITLE
Vscode language extension

### PR DESCRIPTION
**Pull request by eliottparis**
This PR adds a [vscode language extension](https://code.visualstudio.com/api/language-extensions/overview) to highlight `.erbui` and `.erbb` source code in [vscode](https://code.visualstudio.com/).

The extension can be installed during `erbb setup` by specifying the new `--with-vscode-support` option.

erbui_file_screenshot

erbb_file_screenshot

### CODE REVIEW

File: build-system/completion/__init__.py [hunk 1]
```diff
@@ -33,9 +34,9 @@ def install_bootloader ():             return 'bootloader'
...
 def setup ():
    if platform.system () == 'Darwin':
-                                       return 'setup', ZeroOrMore ([with_xcode_support])
+                                       return 'setup', ZeroOrMore ([with_xcode_support, with_vscode_support])
    else:
-                                       return 'setup'
+                                       return 'setup', ZeroOrMore ([with_vscode_support])
 def init ():                           return 'init', ZeroOrMore ([name, language])
 def configure ():                      return 'configure'
 def build ():                          return 'build', [build_simulator, build_firmware, build_hardware]
```
Comment (ohmtech-rdi) [inline build-system/completion/__init__.py:42]: Lower in `DESCRIPTION` the description for `--with-vscode-support` is missing.
Actually also the case for `--with-xcode-support`, so maybe fill free to put this in this PR as well as a typo correction.

```
'--with-xcode-support': 'for Xcode support',
'--with-vscode-support': 'for Visual Studio Code support',
```

File: build-system/scripts/erbb [hunk 2]
```diff
@@ -409,6 +412,11 @@ def parse_args_setup (parent):
          help = 'add Xcode extra support such as erbui/erbb syntax coloring'
       )
 
+   parser.add_argument(
+      '--with-vscode-support',
+      action = 'store_true',
+      help = 'add Xcode extra support such as erbui/erbb syntax coloring'
```
Comment (ohmtech-rdi) [inline build-system/scripts/erbb:418]:
```suggestion
      help = 'add Visual Studio Code extra support such as erbui/erbb syntax coloring'
```

File: build-system/setup/__init__.py [hunk 3]
```diff
@@ -438,6 +439,28 @@ def install_ideplugin (name):
...
+   path_vscode_extensions = os.path.join (Path.home (), '.vscode', 'extensions')
+   path_src = os.path.join (PATH_BUILD_SYSTEM, 'setup', 'vscode')
+   path_dst = os.path.join (path_vscode_extensions, 'eurorack-blocks-vscode')
+
+   if os.path.exists (path_vscode_extensions) and os.path.isdir (path_vscode_extensions):
+      if os.path.exists (path_dst) and os.path.isdir (path_dst):
+         shutil.rmtree (path_dst)
+      shutil.copytree (path_src, path_dst)
+   else:
+      print (f"\033[33mwarning:\033[0m Can't find vscode extensions folder {path_vscode_extensions}.\nPlease make sure vscode is properly installed and run `erbb setup` again.")
```
Comment (ohmtech-rdi) [inline build-system/setup/__init__.py:460]:
```suggestion
      print (f"\033[33mwarning:\033[0m Can't find vscode extensions folder {path_vscode_extensions}.\nPlease make sure vscode is properly installed and run `erbb setup --with-vscode-support` again.")
```

If we forgot to do that as well for Xcode, please don't hesitate to fix in that PR

**Review by ohmtech-rdi — CHANGES_REQUESTED** [review]
Very cool, some changes here and there, but overall good work and a very welcomed addition 👍

File: build-system/completion/__init__.py (hunk 1 above)
Comment (eliottparis) [reply build-system/completion/__init__.py:42]: Will do 👍 , I was not sure if it was missing on purpose or not in the first place for xcode.

File: build-system/scripts/erbb (hunk 2 above)
Comment (eliottparis) [reply build-system/scripts/erbb:418]: of course :)

**Review by ohmtech-rdi — APPROVED** [review]
LGTM 👍 

(quite amazing to see that VS Code can hotload syntax coloring extension 🤯)

**Pull request merged**